UefiCpuPkg/PiSmmCpuDxeSmm/SmmProfile.c

```diff
@@ -438,7 +438,14 @@ InitProtectedMemRange (
          &MemorySpaceMap
          );
   for (Index = 0; Index < NumberOfDescriptors; Index++) {
-    if (MemorySpaceMap[Index].GcdMemoryType == EfiGcdMemoryTypeMemoryMappedIo) {
+    if ((MemorySpaceMap[Index].GcdMemoryType == EfiGcdMemoryTypeMemoryMappedIo) &&
+        ((MemorySpaceMap[Index].BaseAddress & (SIZE_4KB - 1)) == 0) &&
+        ((MemorySpaceMap[Index].Length & (SIZE_4KB - 1)) == 0))

```

Comment:
can you use IS_ALIGNED() macro to check BaseAddress?
Can you simply use "%" to check Length?

Reply:
2nd comment: can you put warning here?

Reply:
Thanks for the comments. I used ADDRESS_IS_ALIGNED for the BaseAddress and "%" for Length. Also put warning here in advance.